server.js

```diff
@@ -3,7 +3,11 @@ const express = require("express");
 const bodyParser = require("body-parser");
 const mongoose = require("mongoose");
 const routes = require("./routes");
+const session = require('express-session');
+const MongoStore = require('connect-mongo')(session);
 const app = express();
+const httpServer = require('http').Server(app);
+const io = require('socket.io')(httpServer);
```

Comment:
This question is a bit in the dark for me, but I found this code snippet online:
```server = http.createServer(app)
io = require('socket.io').listen(server)
require('.sockets')(io);```

And it looks like you follow that structure BUT you don't include `.listen` in this line so it becomes:
`const io = require('socket.io')(httpServer);` instead of `io = require('socket.io').listen(server)`

It might work fine, but I noticed it looked different and might end up mattering.

Here is the stackoverflow question I found with relevant code snippets:
https://stackoverflow.com/questions/23281594/configuring-express-4-0-routes-with-socket-io

Reply:
Yeah later on I just have the `httpServer` listen, which is shared between socketio and express.